### PR TITLE
User creation

### PR DESCRIPTION
**Pull request by JoriPeterson**
- Create first endpoint for application: User Creation
- Add dependencies
- Create migrations for user
- Endpoint is tested and working in Postman

@tayjames

### CODE REVIEW

File: migrations/20191002020117-create-user.js [hunk 1]
```diff
@@ -9,10 +9,15 @@ module.exports = {
         type: Sequelize.INTEGER
       },
       email: {
-        type: Sequelize.STRING
+        type: Sequelize.STRING,
+        allowNull: false,
+        isEmail: true,
+        isUnique: true
```
Comment (tayjames) [inline migrations/20191002020117-create-user.js]: I love what you've done here -- I'm guessing that these are what would be used for validations in Rails? Really cool piece of code that I'll try to implement later in my project if I have the opportunity to do!

File: routes/api/v1/users.js [hunk 2]
```diff
@@ -1,11 +1,28 @@
...
+      password: bcrypt.hashSync(req.body.password, saltRounds),
+      apiKey: uuidv1()
+    })
+    .then(user => {
+    res.status(201).send(JSON.stringify({ apiKey: user.apiKey }))
+    })
+    .catch(error => {
+      res.status(500).send(JSON.stringify("Your information is not correct!"));
+    })
+  } else {
```
Comment (tayjames) [inline routes/api/v1/users.js]: I would love to see this extrapolated a bit more into separate else cases--like if a password doesn't match or if a field is missing! Might be helpful for future devs working on your code base!!

File: test/app.spec.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+describe('POST /api/v1/users', () => {
+  it('User should post to db and api key returned', () => {
+    request(app)
+      .post('/users')
+      .send('email=jori@gmail', 'password=password', 'passwordConfirmation=password')
+      .set('Accept', 'application/json')
+      .expect(function(res) {
+        res.body.apiKey = user.apiKey;
+        })
+      .expect(201);
```
Comment (tayjames) [inline test/app.spec.js]: I like this chaining

File: routes/api/v1/users.js (hunk 2 above)
Comment (JoriPeterson) [reply routes/api/v1/users.js]: I plan to do this when I get a little further into it. I'll keep you posted!

**Pull request merged**